en/docs/chapter_hashing/hash_map.md

```diff
@@ -484,24 +484,26 @@ There are three common ways to traverse a hash table: traversing key-value pairs
 
-## Simple implementation of hash table
+## Simple implementation of a hash table
 
-First, let's consider the simplest case: **implementing a hash table using just an array**. In the hash table, each empty slot in the array is called a <u>bucket</u>, and each bucket can store one key-value pair. Therefore, the query operation involves finding the bucket corresponding to the `key` and retrieving the `value` from it.
+First, let's consider the simplest case: **implementing a hash table using only one array**. In the hash table, each empty slot in the array is called a <u>bucket</u>, and each bucket can store a key-value pair. Therefore, the query operation involves finding the bucket corresponding to the `key` and retrieving the `value` from it.
 
-So, how do we locate the appropriate bucket based on the `key`? This is achieved through a <u>hash function</u>. The role of the hash function is to map a larger input space to a smaller output space. In a hash table, the input space is all possible keys, and the output space is all buckets (array indices). In other words, input a `key`, **and we can use the hash function to determine the storage location of the corresponding key-value pair in the array**.
 
-The calculation process of the hash function for a given `key` is divided into the following two steps:
+So, how do we locate the corresponding bucket based on the `key`? This is achieved through a <u>hash function</u>. The role of the hash function is to map a larger input space to a smaller output space. In a hash table, the input space consists of all the keys, and the output space consists of all the buckets (array indices). In other words, given a `key`, **we can use the hash function to determine the storage location of the corresponding key-value pair in the array**.
 
-1. Calculate the hash value using a certain hash algorithm `hash()`.
-2. Take the modulus of the hash value with the number of buckets (array length) `capacity` to obtain the array index `index`.
+When given a `key`, the calculation process of the hash function consists of the following two steps:
+
+1. Calculate the hash value by using a certain hash algorithm `hash()`.
+2. Take the modulus of the hash value with the bucket count (array length) `capacity` to obtain the array `index` corresponding to that key.
 
 ```shell
 index = hash(key) % capacity
 ```
 
-Afterward, we can use `index` to access the corresponding bucket in the hash table and thereby retrieve the `value`.
+Afterward, we can use the `index` to access the corresponding bucket in the hash table and thereby retrieve the `value`.
+
+Let's assume that the array length is `capacity = 100`, and the hash algorithm is defined as `hash(key) = key`. Therefore, the hash function can be expressed as `key % 100`. The following diagram illustrates the working principle of the hash function using `key` as student ID and `value` as name.
 
```

Comment:
Remove this redundant blank line.

Same as others. Please carefully check the formatting

Reply:
Only one blank line exists before "Afterward..." and "Let's assume..." paragraphs. You could have a double check.